src/classes/RD2_ScheduleService.cls

```diff
@@ -729,4 +770,56 @@ public without sharing class RD2_ScheduleService {
             newSchedulesByRdId.get(rd.RecurringDonation__c).add(rd);
         }
     }
+
+    /**
+    * @description Sets End Date on input rdScheduls.
+    * The end date is not set on a schedule for a fixed-length Recurring Donation.
+    * Thus, schedules should be dynamically evaluated for the End Date
+    * when the Next Donation Date record is being determined, ie the Next Donation Date calculated,
+    * or the Visualizer displays upcoming installments.
+    * @param rd Recurring Donation record
+    * @param rdSchedules Schedules on the Recurring Donation record
+    * @return void
+    */
+    public void evaluateScheduleEndDateForFixedLength(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
```

Comment:
This is the method from PR #5212 that determines the schedule end date dynamically in order to determine the Next Donation Date value and the Opp record. It is used in the Visualizer in PR #5212, but it is not required here to be done or tested.
FYI @sungysang , @andrewyu-salesforce